### PR TITLE
GitHub build workflows

### PR DESCRIPTION
**Pull request by Blixibon**
GitHub workflows which automatically build the codebase on VC++ (Windows) and GCC/G++ (Linux).

The workflows will build for the game DLLs, the shaders, and the mapping tools depending on what files the PR modifies. A manual workflow can be used to compile everything in the SDK.

Based on a script created by z33ky. (https://github.com/mapbase-source/source-sdk-2013/pull/162)

**This PR should be squashed.**

### CODE REVIEW

**Comment by Blixibon** [comment]
@z33ky Out of curiosity, what do you think of the way I implemented this? I used [the script you created](https://github.com/NTKS-Team/source-sdk-2013/blob/91eef577deed1bb28d2f88d68ce0ef22b62879bd/.github/workflows/msbuild.yml) as a base and diversified it to encompass other parts of the SDK, but I don't know if I did it in the most efficient way.

I also implemented a job for building Linux binaries, but I'm not experienced with Linux development and I'm curious if there's anything about the way I did it which could be more efficient or should be addressed. I have not yet tested the generated binaries in-game.

**Review by 1upD — APPROVED** [review]
This looks good to me

File: .github/workflows/mapbase_build-base.yml [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+    - name: Enable VS2022
+      working-directory: '${{inputs.branch}}/src/vpc_scripts'
+      shell: bash
+      run: sed -i 's/^\($Conditional[ 	]\+VS2022[ 	]\+\).*/\1"1"/' newer_vs_toolsets.vpc
+
+    - name: Pick game
+      if: inputs.project-group == 'game' || inputs.project-group == 'shaders'
+      working-directory: '${{inputs.branch}}/src'
+      shell: bash
+      run: sed -i 's/\/hl2 \/episodic/\/${{inputs.game}}/' create${{inputs.project-group}}projects.bat
```
Comment (z33ky) [inline .github/workflows/mapbase_build-base.yml:68]: Note you can use a different separator for sed, like `sed -i 's|/hl2 /episodic|/${{inputs.game}}|'` which is more easily readable by skipping the need to escape the slashes.

**Review by z33ky — COMMENTED** [review]
Hi. Very cool.
Sorry for taking some time to reply.

The generated Linux binaries run fine.

You could consider also publishing the debug-info files for release builds as they are stripped off the binaries. On Linux they are also put in the `bin/` directory with `.dbg` suffix (i.e. `client.so.dbg`). I remember on Windows they were `.pdb` files.
On Linux at least the debug-builds do not strip the debug-info from the libraries, so publishing the .dbg files there is unnecessary.

**Pull request merged**